### ISSUE TITLE
Export the inner insets

### ISSUE THREAD

**Issue by jayasme** [react-native-community/react-native-safe-area-view#22]
Hi, I am doing a job that is to make a `TextInput` keep moving with the keyboard.

I've finished the job on no-iPhone X devices, it works perfect, but when I switched to the iPhone X with `SafeAreaView` I found a problem.

<img width="400" alt="1" src="https://user-images.githubusercontent.com/668198/37509259-98b2d054-2931-11e8-866e-96ecde1f1e54.png">

Here is what it looked like when the keyboard collapsed, it was not the problem.

<img width="403" alt="2" src="https://user-images.githubusercontent.com/668198/37509362-079a17fc-2932-11e8-93fd-188bccf090ab.png">

Here is what it looked like when the keyboard displayed, you can see the inset of the `SafeArewView` was still existing in the text area.

I've tried not to use `forceInsets` to the `SafeAreaView`, unfortunately, it didn't work.

So my solution is subtracting the offset by the bottom inset of the `SafeAreaView` which equals to 34 on the iPhone X, however, it's too tricky to subtract it by 34 directly, alternatively, I want the `SafeAreaView` to tell me how many the inset is, this is why I want `SafeAreaView` to export the calculated inner insets.

I know in the most of cases, the exporting is not necessary and should not be abused, but I really need these insets in this situation and I believe there must few situations that can only be resolved with the insets directly, please care about my suggestion. Thank you!

**Comment by dylancom** [comment]
I'm also in the need for the inset sizes.

**Comment by JulianKingman** [comment]
I'd love to have access to the values without having to render the view, so I can do my own custom UI logic with the values.

**Comment by digital-flowers** [comment]
I believe this can solve your issue and it will be smooth:
```javascript
import {Keyboard} from "react-native";

export class AddMessageBox extends Component {

  state = {
    hasKeyboard: false
  };

  componentDidMount() {
    this.keyboardDidShowListener = Keyboard.addListener('keyboardWillShow', () => this.setState({hasKeyboard: true}));
    this.keyboardDidHideListener = Keyboard.addListener('keyboardWillHide', () => this.setState({hasKeyboard: false}));
  }

  componentWillUnmount() {
    this.keyboardDidShowListener.remove();
    this.keyboardDidHideListener.remove();
  }

  render() {
     const {hasKeyboard} = this.state;
    return (
      <SafeAreaView forceInset={{top: "never", bottom: hasKeyboard ? 0 : "always"}}>
         ......
      </SafeAreaView>
   )
}
```

**Comment by Elijen** [comment]
I've created a PR that allows you to use SafeAreaView's `ref` to access the insets (you still need to render the component though).

https://github.com/react-native-community/react-native-safe-area-view/pull/53

**Comment by jfrolich** [comment]
Would it be possible to export the default insets for top, bottom etc. statically (based on device type). I need it to have a SafeAreaView that animates the inset when the keyboard is appearing.

**Comment by brentvatne** [comment]
this should be resolved in version 1.0 which was released today, thanks for reporting!
